QuarkusTestExtension fails to instantiate test class: java.lang.VerifyError in enhanced Entity class

**Describe the bug**
I have converted the Spring PetClinic domain model to Quarkus and created a simple repository and REST resource; both are working fine. After writing a Mock repository I wrote an initial QuarkusTest for the REST resource. 

Running the test produces a `java.lang.VerifyError` in the enhanced entity:

```
Caused by: java.lang.VerifyError:
Bad type on operand stack
Exception Details:
  Location:
    com/github/acme/quarkus/petclinic/model/Pet.$$_hibernate_write_visits(Ljava/util/Set;)V @188: invokevirtual
  Reason:
    Type 'com/github/acme/quarkus/petclinic/model/Pet' (current frame, stack[1]) is not assignable to 'java/lang/Long'
  Current Frame:
    bci: @188
    flags: { }
    locals: { 'com/github/acme/quarkus/petclinic/model/Pet', 'java/util/Set', '[Ljava/lang/Object;', integer }
    stack: { 'com/github/acme/quarkus/petclinic/model/Visit', 'com/github/acme/quarkus/petclinic/model/Pet' }
```

**Expected behavior**
A failing unit test, which can be fixed by adding the correct JSON response

**Actual behavior**
JVM bytecode crash


**To Reproduce**
Steps to reproduce the behavior:
1. Run `mvn compile quarkus:dev`
2. Test using HTTPie: `http :8080/vets` works!
3. Stop Quarkus dev
4. Run `mvn test` produces the error

**Environment (please complete the following information):**
 - Windows / MacOS
 - OpenJDK Runtime Environment (build 1.8.0_222-20190711112428.buildslave.jdk8u-src-tar--b08)
- OpenJDK 64-Bit GraalVM CE 19.2.0 (build 25.222-b08-jvmci-19.2-b02, mixed mode)



** Output test run **
```
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running com.github.acme.quarkus.petclinic.web.resource.VetResourceTest
[INFO] H2 database started in TCP server mode
12:35:29,531 INFO  [io.qua.dep.QuarkusAugmentor] Beginning quarkus augmentation
12:35:30,282 INFO  [org.jbo.threads] JBoss Threads version 3.0.0.Beta5
12:35:31,098 INFO  [org.hib.jpa.boo.int.PersistenceXmlParser] HHH000318: Could not find any META-INF/persistence.xml file in the classpath
12:35:31,783 INFO  [org.hib.Version] HHH000412: Hibernate Core {5.4.4.Final}
12:35:32,814 INFO  [io.qua.dep.QuarkusAugmentor] Quarkus augmentation completed in 3283ms
12:35:33,495 INFO  [org.hib.byt.enh.int.byt.BiDirectionalAssociationHandler] Could not find bi-directional association for field [com.github.acme.quarkus.petclinic.model.Pet#type]
12:35:34,164 INFO  [org.hib.ann.com.Version] HCANN000001: Hibernate Commons Annotations {5.1.0.Final}
12:35:34,424 INFO  [org.hib.dia.Dialect] HHH000400: Using dialect: org.hibernate.dialect.H2Dialect
[INFO] H2 database started in TCP server mode
12:35:34,945 INFO  [io.qua.dep.QuarkusAugmentor] Beginning quarkus augmentation
12:35:35,651 INFO  [org.hib.jpa.boo.int.PersistenceXmlParser] HHH000318: Could not find any META-INF/persistence.xml file in the classpath
[INFO] H2 database was shut down
[ERROR] Tests run: 2, Failures: 0, Errors: 2, Skipped: 0, Time elapsed: 9.789 s <<< FAILURE! - in com.github.acme.quarkus.petclinic.web.resource.VetResourceTest
[ERROR] testVetsEndpoint  Time elapsed: 0.008 s  <<< ERROR!
org.junit.jupiter.api.extension.TestInstantiationException: TestInstanceFactory [io.quarkus.test.junit.QuarkusTestExtension] failed to instantiate test class [com.github.acme.quarkus.petclinic.web.resource.VetResourceTest]
Caused by: java.lang.ExceptionInInitializerError
Caused by: java.lang.RuntimeException: Failed to start quarkus
Caused by: java.lang.VerifyError:
Bad type on operand stack
Exception Details:
  Location:
    com/github/acme/quarkus/petclinic/model/Pet.$$_hibernate_write_visits(Ljava/util/Set;)V @188: invokevirtual
  Reason:
    Type 'com/github/acme/quarkus/petclinic/model/Pet' (current frame, stack[1]) is not assignable to 'java/lang/Long'
  Current Frame:
    bci: @188
    flags: { }
    locals: { 'com/github/acme/quarkus/petclinic/model/Pet', 'java/util/Set', '[Ljava/lang/Object;', integer }
    stack: { 'com/github/acme/quarkus/petclinic/model/Visit', 'com/github/acme/quarkus/petclinic/model/Pet' }
  Bytecode:
    0x0000000: 2ab4 0004 c600 442a b400 0413 01e2 b801
    0x0000010: d299 0037 2ab4 0004 b901 e601 004d 033e
    0x0000020: 1d2c bea2 0025 2bc6 000f 2b2c 1d32 b901
    0x0000030: dd02 009a 000f 2c1d 3201 57c0 006b 01b6
    0x0000040: 01e9 8403 01a7 ffdb a700 032a 4d2b 4e2c
    0x0000050: b601 b0c6 001f 2c2c b601 b02c 1301 e12c
    0x0000060: b400 042d b901 c505 00c0 0068 b500 04a7
    0x0000070: 000e 2c2d c000 68b5 0004 a700 032b c600
    0x0000080: 472b 1301 e2b8 01d2 9900 3d2b b901 e601
    0x0000090: 004d 033e 1d2c bea2 002e 2c1d 3213 01e2
    0x00000a0: b801 d299 001c 2c1d 32c0 006b b601 ed2a
    0x00000b0: a500 0f2c 1d32 2a57 c000 6b2a b601 e984
    0x00000c0: 0301 a7ff d2a7 0003 b1
  Stackmap Table:
    append_frame(@32,Object[#495],Integer)
    same_frame(@54)
    same_frame(@66)
    chop_frame(@72,2)
    same_frame(@75)
    append_frame(@114,Object[#17],Object[#104])
    same_frame(@125)
    full_frame(@148,{Object[#17],Object[#104],Object[#495],Integer},{})
    same_frame(@191)
    full_frame(@197,{Object[#17],Object[#104],Object[#480]},{})
    same_frame(@200)


[ERROR] unittestFindVetByNameEndpoint  Time elapsed: 0.003 s  <<< ERROR!
org.junit.jupiter.api.extension.TestInstantiationException:
TestInstanceFactory [io.quarkus.test.junit.QuarkusTestExtension] failed to instantiate test class [com.github.acme.quarkus.petclinic.web.resource.VetResourceTest]: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor#build threw an exception: io.quarkus.deployment.configuration.ConfigurationError: Unable to properly register the hierarchy of the following JPA classes as they are not in the Jandex index:
        - org.hibernate.engine.spi.ManagedMappedSuperclass
Consider adding them to the index either by creating a Jandex index for your dependency via the Maven plugin, an empty META-INF/beans.xml or quarkus.index-dependency properties.
Caused by: java.lang.RuntimeException:
io.quarkus.builder.BuildException: Build failure: Build failed due to errors
        [error]: Build step io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor#build threw an exception: io.quarkus.deployment.configuration.ConfigurationError: Unable to properly register the hierarchy of the following JPA classes as they are not in the Jandex index:
        - org.hibernate.engine.spi.ManagedMappedSuperclass
Consider adding them to the index either by creating a Jandex index for your dependency via the Maven plugin, an empty META-INF/beans.xml or quarkus.index-dependency properties.
Caused by: io.quarkus.builder.BuildException:
Build failure: Build failed due to errors
        [error]: Build step io.quarkus.hibernate.orm.deployment.HibernateOrmProcessor#build threw an exception: io.quarkus.deployment.configuration.ConfigurationError: Unable to properly register the hierarchy of the following JPA classes as they are not in the Jandex index:
        - org.hibernate.engine.spi.ManagedMappedSuperclass
Consider adding them to the index either by creating a Jandex index for your dependency via the Maven plugin, an empty META-INF/beans.xml or quarkus.index-dependency properties.
Caused by: io.quarkus.deployment.configuration.ConfigurationError:
Unable to properly register the hierarchy of the following JPA classes as they are not in the Jandex index:
        - org.hibernate.engine.spi.ManagedMappedSuperclass
Consider adding them to the index either by creating a Jandex index for your dependency via the Maven plugin, an empty META-INF/beans.xml or quarkus.index-dependency properties.

[INFO]
[INFO] Results:
[INFO]
[ERROR] Errors:
[ERROR]   VetResourceTest.testVetsEndpoint � TestInstantiation TestInstanceFactory [io.q...
[ERROR]   VetResourceTest.unittestFindVetByNameEndpoint � TestInstantiation TestInstance...
[INFO]
[ERROR] Tests run: 2, Failures: 0, Errors: 2, Skipped: 0
```

--- a/bom/runtime/pom.xml
+++ b/bom/runtime/pom.xml
@@ -82,7 +82,7 @@
         <commons-codec.version>1.13</commons-codec.version>
         <classmate.version>1.3.4</classmate.version>
         <hibernate-validator.version>6.1.0.Final</hibernate-validator.version>
-        <hibernate-orm.version>5.4.7.Final</hibernate-orm.version>
+        <hibernate-orm.version>5.4.8.Final</hibernate-orm.version>
         <hibernate-search.version>6.0.0.Beta2</hibernate-search.version>
         <narayana.version>5.9.8.Final</narayana.version>
         <jboss-transaction-api_1.2_spec.version>1.1.1.Final</jboss-transaction-api_1.2_spec.version>

